alf/networks/encoding_networks_test.py

```diff
@@ -40,28 +40,33 @@ def test_empty_layers(self):
         self.assertEmpty(network._fc_layers)
         self.assertTrue(network._img_encoding_net is None)
 
-    @parameterized.parameters((True, ), (False, ))
-    def test_image_encoding_network(self, flatten_output):
+    @parameterized.parameters((True, False), (False, True))
+    def test_image_encoding_network(self, flatten_output, same_padding):
         input_spec = TensorSpec((3, 32, 32), torch.float32)
         img = input_spec.zeros(outer_dims=(1, ))
         network = ImageEncodingNetwork(
             input_channels=input_spec.shape[0],
             input_size=input_spec.shape[1:],
             conv_layer_params=((16, (2, 2), 1, (1, 0)), (15, 2, (1, 2), 1)),
+            same_padding=same_padding,
             activation=torch.tanh,
             flatten_output=flatten_output)
 
         self.assertLen(list(network.parameters()), 4)  # two conv2d layers
 
         output, _ = network(img)
-        output_shape = (15, 34, 16)
+        if same_padding:
+            output_shape = (15, 34, 17)
```

Comment:
This doesn't like same_padding. If same spatial dimension should be (32, 16)

Reply:
"same" padding is a wrong word. It's essentially taking a padding=kernel//2. If kernel is an odd number, then one additional padding is applied to the right. when strides=1, the output size will be the same; if not, it will not be the same output size. See my comments in ImageEncodingNetwork

Reply:
Made a minor change so that now if kernel size is an odd number and strides=1, then the output size will be the same. Note that because PyTorch doesn't support different paddings for left and right, it's impossible to still get the same output size if kernel size is an even number, by just configuring PyTorch conv2d or transposedconv2d layers. https://pytorch.org/docs/stable/nn.html#torch.nn.Conv2d